src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/SelfHostDeployer.cs

```diff
@@ -189,5 +192,62 @@ public override void Dispose()
                 StopTimer();
             }
         }
+
+        // Temporary workaround for https://github.com/dotnet/cli/issues/6286
+        private void ProvideExecutePermission()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                && DeploymentParameters.RuntimeFlavor == RuntimeFlavor.CoreClr
+                && DeploymentParameters.ApplicationType == ApplicationType.Standalone)
+            {
+                var executablePath = Path.Combine(
+                    DeploymentParameters.PublishedApplicationRootPath,
+                    DeploymentParameters.ApplicationName);
+
+                var chmodProcessStartInfo = new ProcessStartInfo
+                {
+                    FileName = "chmod",
+                    Arguments = $"755 {executablePath}",
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                };
+
+                var chmodProcess = new Process() { StartInfo = chmodProcessStartInfo };
+                try
+                {
+                    chmodProcess.StartAndCaptureOutAndErrToLogger("chmod", Logger);
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError(
+                        "Error occurred while starting the process '{processName}'. Exception: {exception}",
+                        chmodProcessStartInfo.FileName,
+                        ex.ToString());
+                }
+
+                var hasExited = chmodProcess.WaitForExit(3 * 1000);
+
+                if (hasExited)
+                {
+                    if (chmodProcess.ExitCode != 0)
+                    {
+                        Logger.LogError(
+                            "Process {processName} with id {pid} exited with code {exitCode} or failed to start.",
+                            chmodProcessStartInfo.FileName,
+                            HostProcess.Id,
+                            HostProcess.ExitCode);
+                    }
+                }
+                else
+                {
+                    Logger.LogError(
+                            "Process {processName} with id {pid} did not exit in the given amount of time.",
```

Comment:
So in this case, I don't mind either way because we control the logger, but in general when logging we should **avoid** string interpolation for cases like this. Logging has special support for these named format arguments and loggers that preserves the key-value mapping between the field name and value, so that structured logging sources can keep the data around.